server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1044,10 +1044,10 @@ private RateLimiter getRateLimiter(Settings repositorySettings, String setting,
     }
 
     protected void assertSnapshotOrGenericThread() {
-        assert Thread.currentThread().getName().contains(ThreadPool.Names.SNAPSHOT)
-            || Thread.currentThread().getName().contains(ThreadPool.Names.GENERIC)
-            || Thread.currentThread().getName().contains(ThreadPool.Names.SEARCH) :
-                "Expected current thread [" + Thread.currentThread() + "] to be the snapshot or generic thread.";
+        assert Thread.currentThread().getName().contains('[' + ThreadPool.Names.SNAPSHOT + ']')
+               || Thread.currentThread().getName().contains('[' + ThreadPool.Names.SEARCH + ']')
```

Comment:
es commit removes the SEARCH thread, but we have to keep it, since we are using `SELECT ... FROM sys.snapshots` in the tests, WHERE a search thread is used.